ci-operator/step-registry/ipi/conf/aws/user-min-permissions/ipi-conf-aws-user-min-permissions-commands.sh

```diff
@@ -285,12 +280,22 @@ EOF
 
 		openshift-install create permissions-policy --dir ${dir}
 
-		# Save policy to shared dir so later steps have access to it.
-		mv ${dir}/${USER_POLICY_FILENAME} ${USER_POLICY_FILE}
+		# Save policy to artifact dir for debugging
+		mv ${dir}/${USER_POLICY_FILENAME} ${ARTIFACT_DIR}/${USER_POLICY_FILENAME}.original.json
+
+		# AWS: the policy created by permissions-policy may exceed the 6144 limition
+		# https://issues.redhat.com/browse/OCPBUGS-45612
+		# Merge multi-Sid into one Sid
+		cat ${ARTIFACT_DIR}/${USER_POLICY_FILENAME}.original.json | jq -r '.Statement[].Action[]' | sort | uniq > ${PERMISION_LIST}
```

Comment:
So the root cause of `OCPBUGS-45612` is that there are several duplicated action items in the policy file generated by the installer ?

Reply:
it generates multi Sids, like:
```
    {
      "Sid": "CreateInstanceRole",
      "Effect": "Allow",
      "Action": [
        "iam:CreateRole",
        "iam:DeleteRole",
        "iam:DeleteRolePolicy",
        "iam:PutRolePolicy"
      ],
      "Resource": [
        "*"
      ]
    },
    {
      "Sid": "CreateInstanceProfile",
      "Effect": "Allow",
      "Action": [
        "iam:AddRoleToInstanceProfile",
        "iam:CreateInstanceProfile",
        "iam:DeleteInstanceProfile",
        "iam:RemoveRoleFromInstanceProfile"
      ],
      "Resource": [
        "*"
      ]
    },
    {
      "Sid": "PermissionDefaultZones",
      "Effect": "Allow",
      "Action": [
        "ec2:DescribeAvailabilityZones",
        "ec2:DescribeInstanceTypeOfferings"
      ],
      "Resource": [
        "*"
      ]
    },
```

to reduce the characters number, we merge them into one Sid.

Reply:
> several duplicated action items

Hmm, it's not _duplicated action items_,  but `Sid` `Effect` `Resource` sections,  they take up too many characters.

Reply:
before https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/58439/rehearse-58439-periodic-ci-openshift-openshift-tests-private-release-4.18-multi-nightly-aws-ipi-byo-kms-etcd-encryption-mini-perm-arm-f2/1864607710884925440/artifacts/aws-ipi-byo-kms-etcd-encryption-mini-perm-arm-f2/ipi-conf-aws-user-min-permissions/artifacts/aws-permissions-policy-creds.json.original.json

after https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/58439/rehearse-58439-periodic-ci-openshift-openshift-tests-private-release-4.18-multi-nightly-aws-ipi-byo-kms-etcd-encryption-mini-perm-arm-f2/1864607710884925440/artifacts/aws-ipi-byo-kms-etcd-encryption-mini-perm-arm-f2/ipi-conf-aws-user-min-permissions/artifacts/aws-permissions-policy-creds.json